src/components/Project/ImageSelect.tsx

```diff
@@ -5,38 +5,74 @@ import { Label } from '@/components/ui/label'
 import { useProjectProvider } from '@/context/ProjectProvider'
 import { uploadFile } from '@/lib/ipfs/uploadToIpfs'
 import { useState } from 'react'
-import AttestButton from './AttestButton'
+import { Button } from '../ui/button'
+import usePaymasterAttest from '@/hooks/project/usePaymasterAttest'
+import { toast } from '../ui/use-toast'
 
 const ImageSelect = () => {
-  const { setImage } = useProjectProvider()
+  const {
+    setImage,
+    uploadingImage,
+    setUploadingImage,
```

Comment:
are these `uploading` state variables used anywhere other than `<ImageSelect />`?
- if no, could just be saved in `state` for this local component right?
- optional.